dev/pyRevitLabs.PyRevit.Runtime/ScriptConsole.cs

```diff
@@ -835,14 +835,33 @@ private System.Windows.Shapes.Path GetAutoCollapseIcon(bool active) {
         }
 
         private void Save_Contents_Button_Clicked(object sender, RoutedEventArgs e) {
-            var saveDlg = new System.Windows.Forms.SaveFileDialog() {
+            var saveDlg = new System.Windows.Forms.SaveFileDialog() 
+            {
                 Title = "Save Output to:",
                 Filter = "HTML|*.html"
+                DefaultExt = "html",
+                AddExtension = true,
+                RestoreDirectory = true
             };
-            saveDlg.ShowDialog();
-            var f = File.CreateText(saveDlg.FileName);
-            f.Write(GetFullHtml());
-            f.Close();
+            if (saveDlg.ShowDialog() == System.Windows.Forms.DialogResult.OK && !string.IsNullOrWhiteSpace(saveDlg.FileName))
+            {
+                try
+                {
+                    using (var f = File.CreateText(saveDlg.FileName))
+                    {
+                        f.Write(GetFullHtml());
+                    }
+                }
+                catch (Exception ex)
+                {
+                    // Use WPF MessageBox instead of System.Windows.Forms.MessageBox
+                    System.Windows.MessageBox.Show($"Error saving file: {ex.Message}", "Save Error", MessageBoxButton.OK, MessageBoxImage.Error);
+                }
+            }
+            else
+            {
+                System.Windows.MessageBox.Show("File saving was canceled or an invalid file name was provided.", "Save Canceled", MessageBoxButton.OK, MessageBoxImage.Information);
```

Comment:
I wouldn't annoy the user with a message box, he/she already know that it canceled it.
just invert the if condition on line 846 and return early (this also avoids excessive nesting in the code)